src/plugins/intel_gna/layers/gna_permute.hpp

```diff
@@ -117,4 +117,30 @@ inline std::vector<int> GetPermuteOrder(InferenceEngine::Layout in_layout, Infer
     return {0, 1, 2, 3};
 }
 
+inline bool isTrivialPermute(const std::vector<int64_t> order, const std::vector<size_t>& input_shape) {
+    // cases when all permutations happened either between 1 and X shape where no other dims in between
+    auto transpose_seq = genPermutations(order.begin(), order.end());
```

Comment:
As a side not I think we should be more consistent in using transpose vs permute, i.e. use permute only in cases when it pertains to legacy permute layer.

Reply:
> As a side not I think we should be more consistent in using transpose vs permute, i.e. use permute only in cases when it pertains to legacy permute layer.

Sure. But I treat this part of code as legacy one.